src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItemSource.cs

```diff
@@ -17,266 +17,247 @@
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
+
+internal sealed class SourceGeneratedFileItemSource(SourceGeneratorItem parentGeneratorItem, Workspace workspace, IAsynchronousOperationListener asyncListener, IThreadingContext threadingContext) : Shell.IAttachedCollectionSource, ISupportExpansionEvents
 {
-    internal class SourceGeneratedFileItemSource : Shell.IAttachedCollectionSource, ISupportExpansionEvents
+    private readonly SourceGeneratorItem _parentGeneratorItem = parentGeneratorItem;
+    private readonly Workspace _workspace = workspace;
+    private readonly IAsynchronousOperationListener _asyncListener = asyncListener;
+    private readonly IThreadingContext _threadingContext = threadingContext;
+
+    /// <summary>
+    /// The returned collection of items. Can only be mutated on the UI thread, as other parts of WPF are subscribed to the change
+    /// events and expect that.
+    /// </summary>
+    private readonly BulkObservableCollectionWithInit<BaseItem> _items = [];
+
+    /// <summary>
+    /// Gate to guard mutation of <see cref="_resettableDelay"/>.
+    /// </summary>
+    private readonly object _gate = new();
+
+    private readonly CancellationSeries _cancellationSeries = new();
+    private ResettableDelay? _resettableDelay;
+
+    public object SourceItem => _parentGeneratorItem;
+
+    // Since we are expensive to compute, always say we have items.
+    public bool HasItems => true;
+
+    public IEnumerable Items => _items;
+
+    private async Task UpdateSourceGeneratedFileItemsAsync(Solution solution, CancellationToken cancellationToken)
     {
-        private readonly SourceGeneratorItem _parentGeneratorItem;
-        private readonly Workspace _workspace;
-        private readonly IAsynchronousOperationListener _asyncListener;
-        private readonly IThreadingContext _threadingContext;
-
-        /// <summary>
-        /// The returned collection of items. Can only be mutated on the UI thread, as other parts of WPF are subscribed to the change
-        /// events and expect that.
-        /// </summary>
-        private readonly BulkObservableCollectionWithInit<BaseItem> _items;
-
-        /// <summary>
-        /// Gate to guard mutation of <see cref="_resettableDelay"/>.
-        /// </summary>
-        private readonly object _gate = new object();
-
-        private readonly CancellationSeries _cancellationSeries = new();
-        private ResettableDelay? _resettableDelay;
-
-        public SourceGeneratedFileItemSource(SourceGeneratorItem parentGeneratorItem, Workspace workspace, IAsynchronousOperationListener asyncListener, IThreadingContext threadingContext)
+        var project = solution.GetProject(_parentGeneratorItem.ProjectId);
+
+        if (project == null)
         {
-            // Construction of BulkObservableCollection requires the main thread
```

Comment:
>// Construction of BulkObservableCollection requires the main thread [](http://example.com/codeflow?start=12&length=68)

Is this not something needing checked anymore?

Reply:
i just don't care :)